### ISSUE TITLE
calendar applet time zone problem

### ISSUE THREAD

**Issue by ZeckeSZ** [linuxmint/cinnamon#7897]
```
 * Cinnamon version: ?
 * Distribution: Mint 18.*, 19, *?
 * Graphics hardware *and* driver used
 * 32 or 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**
Calendar applet: if I select time zone "atlantic/azores", the calendar always shows "sunday" as the first day of the week, even if something else was selected.
If I select "sunday" as the first day, "saturday" will be displayed as the first day in the calendar.
This also shifts the days, correct would be Friday, shown is Thursday.

**Steps to reproduce**
change time zone to "atlantic/azores"

**Expected behaviour**
see above

**Other information**

**Comment by icarter09** [comment]
@ZeckeSZ where exactly are you changing the time zone selection?

**Comment by ZeckeSZ** [comment]
This way:
Click on panel calendar applet  --> Date and Time Settings --> Unlock --> Region = Atlantic, City = Azores, then just restart cinnamon

**Comment by icarter09** [comment]
@ZeckeSZ If I do not do the "restart cinnamon" step, then the calendar applet works as intended. But when I restart Cinnamon, I see the days get shifted as you described.

**Comment by Woodley** [comment]
I am running the Cinnamon 19.0 Live on USB and change the Time zone to Central USA (America, Chicago) using the Date and Time Settings when clicking on the date time in menu. I turn off show 24h, turn on display date and display seconds. 

Day of week shifts, and tomorrow is highlighted. See Photo Attached.
![bug - calendar day of week shift](https://user-images.githubusercontent.com/17752792/48273184-f7e8d200-e405-11e8-9374-824972140a75.jpg)
This one shows Month Shift instead of day of week, probably due to tomorrow being highlighted:
![bug - calendar month shift](https://user-images.githubusercontent.com/17752792/48273564-d805de00-e406-11e8-9c36-8bd3600a8805.jpg)

**Comment by emilieroberts** [comment]
Seeing similar things. Notice:
- Today is Sunday Jan 26. Calendar applet shows it as Sun Jan 25. The week labelling should also start on sunday, not monday.
![calendar-jan26](https://user-images.githubusercontent.com/8290920/73143253-2d73ec80-4098-11ea-91fe-b85d7097d8be.png)
- Thinking the week starting/locale dates were the issue, I changed the settings to always start on monday. The result was that the week labelling now started with Tuesday. The shown calendar for May is wrong: May 11 2020 is a monday not a tuesday. May 14 2020 is a thursday, not a friday
![calendar-labels start on tues-small](https://user-images.githubusercontent.com/8290920/73143257-3bc20880-4098-11ea-85e8-3bcf143772b4.png)
- See settings for the second screenshot 
![calendar-settings](https://user-images.githubusercontent.com/8290920/73143256-349afa80-4098-11ea-8111-3c78c77b7433.png)

To get into this state: I was in San Francisco with my time zone set to Los Angeles. I then traveled to Paris and updated my time zone to Paris. Network time is enabled.

Rebooting the computer fixed the issue with the applet.
